mezzanine/core/managers.py

```diff
@@ -19,6 +21,34 @@
 from mezzanine.utils.urls import home_slug
 
 
+if django.VERSION >= (1, 10):
+    class ManagerDescriptor(ManagerDescriptor):
+        """
+        This class exists purely to skip the abstract model check
+        in the __get__ method of Django's ManagerDescriptor.
+        """
+        def __get__(self, instance, cls=None):
+            if instance is not None:
+                raise AttributeError(
+                    "Manager isn't accessible via %s instances" % cls.__name__
+                )
+
+            # In ManagerDescriptor.__get__, an exception is raised here
+            # if cls is abstract
+
+            if cls._meta.swapped:
+                raise AttributeError(
+                    "Manager isn't available; "
+                    "'%s.%s' has been swapped for '%s'" % (
+                        cls._meta.app_label,
+                        cls._meta.object_name,
+                        cls._meta.swapped,
+                    )
+                )
+
+            return cls._meta.managers_map[self.manager.name]
+
+
```

Comment:
This seems to work but it's very ugly. It might be worth creating a dedicated search API which doesn't use the Manager convention, and to deprecate this?